app/views/admin/editions/_appointment_fields.html.erb

```diff
@@ -0,0 +1,20 @@
+<% cache_if edition.role_appointment_ids.empty?, taggable_ministerial_role_appointments_cache_digest do %>
```

Comment:
We may need to tweak how this caching works. Right now it looks like `appointment_fields` and `legacy_appointment_fields` will both cache to the same key.

So I think there's a good chance the rendered list will end up randomly switching between new and old design systems due to the cache key clash.

```suggestion
<% cache_if edition.role_appointment_ids.empty?, "#{taggable_ministerial_role_appointments_cache_digest}-design-system" do %>
```

Something like this might be the solution – just appending something to the end of the cache key so they don't clash.